support-workers/src/test/scala/com/gu/support/workers/DigitalSubscriptionGiftRedemptionSpec.scala

```diff
@@ -0,0 +1,75 @@
+package com.gu.support.workers
+
+import com.gu.support.SerialisationTestHelpers
+import com.gu.support.workers.lambdas.DigitalSubscriptionGiftRedemption
+import com.gu.support.workers.lambdas.DigitalSubscriptionGiftRedemption.{Expired, Redeemed, RedeemedInThisRequest, Unredeemed}
+import com.gu.support.zuora.api.response.{SubscriptionRedemptionFields, SubscriptionRedemptionQueryResponse}
+import org.joda.time.LocalDate
+import org.scalatest.flatspec.AnyFlatSpec
+
+class DigitalSubscriptionGiftRedemptionSpec extends AnyFlatSpec with SerialisationTestHelpers {
+  "SubscriptionRedemptionQueryResponse" should "deserialise correctly" in {
+    val jsonResponse = """
+        {
+        "records": [
+            {
+                "CreatedRequestId__c": "35b4c314-d982-4386-983e-2e8c453f50be",
+                "Id": "2c92c0f8742dcaf5017434d002e73a56",
+                "ContractEffectiveDate": "2020-08-09"
+            }
+        ],
+        "done": true,
+        "size": 1
+    }
+    """
+
+    testDecoding[SubscriptionRedemptionQueryResponse](
+      jsonResponse,
+      subResponse => subResponse.records.head.contractEffectiveDate.getDayOfMonth shouldBe 9
+    )
+  }
```

Comment:
not a major thing but does this belong in a different test class? This one is mostly testing `getSubscriptionState` which is nice and clear seeing all the different variations on their own.